File: seed/php-model/literal/src/Inlined/ATopLevelLiteral.php
<?php

namespace Seed\Inlined;

use Seed\Core\SerializableType;
use Seed\Core\JsonProperty;

class ATopLevelLiteral extends SerializableType
{
    /**
     * @var ANestedLiteral $nestedLiteral
     */
    #[JsonProperty("nestedLiteral")]
    public ANestedLiteral $nestedLiteral;

    /**
     * @param ANestedLiteral $nestedLiteral
     */
    public function __construct(
        ANestedLiteral $nestedLiteral,
    ) {
        $this->nestedLiteral = $nestedLiteral;
    }
}

File: seed/pydantic-v2/literal/resources/inlined/types/a_top_level_literal.py
from pydantic import BaseModel
from resources.inlined.types.a_nested_literal import ANestedLiteral
from dt import datetime
from core.datetime_utils import serialize_datetime


class ATopLevelLiteral(BaseModel):
    nested_literal: ANestedLiteral

    class Config:
        frozen = True
        smart_union = True
        json_encoders = {datetime: serialize_datetime}
